apps/src/submit.js

```diff
@@ -0,0 +1,115 @@
+import commonMsg from '@cdo/locale';
+import dom from './dom';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import DialogButtons from './templates/DialogButtons';
+
+var Submit = module.exports = function (options) {
+  this.studioApp = options.studioApp;
+  this.onPuzzleComplete = options.onPuzzleComplete.bind(this);
+  this.unsubmitUrl = options.unsubmitUrl;
+};
+
+// The DOM should have submitButton and unsubmitButton.  Set up handlers for them.
+Submit.prototype.setupButtons = function () {
+  var submitButton = document.getElementById('submitButton');
+  if (submitButton) {
+    dom.addClickTouchEvent(submitButton, this.onPuzzleSubmit.bind(this));
+  }
+
+  var unsubmitButton = document.getElementById('unsubmitButton');
+  if (unsubmitButton) {
+    dom.addClickTouchEvent(unsubmitButton, this.onPuzzleUnsubmit.bind(this));
+  }
+};
+
+// If submit succeeds, then redirect appropriately.
+Submit.prototype.onSubmitComplete = function (response) {
+  window.location.href = response.redirect;
+};
+
+// Called when the user confirms they want to submit this solution.
+Submit.prototype.onPuzzleSubmitConfirm = function () {
+  this.onPuzzleComplete(true);
+};
```

Comment:
This method seems unnecessary. Could just have `onConfirm: this.onPuzzleComplete(true).bind(this)` on line 41?